faker/providers/ssn/en_US/__init__.py

```diff
@@ -1,19 +1,171 @@
+
+        itin = "{0:03d}-{1:02d}-{2:04d}".format(area, group, serial)
+        return itin
+
+    def ein(self):
+        """Generate a random United States Employer Identification Number (EIN).
+        
+         An United States An Employer Identification Number (EIN) is
+         also known as a Federal Tax Identification Number, and is
+         used to identify a business entity. EINs follow a format of a
+         two-digit prefix followed by a hyphen and a seven-digit sequence:
+         ##-######
+
+         https://www.irs.gov/businesses/small-businesses-self-employed/employer-id-numbers
+        """
+
+        # Only certain EIN Prefix values are assigned:
+        #
+        # https://www.irs.gov/businesses/small-businesses-self-employed/how-eins-are-assigned-and-valid-ein-prefixes
+
+        ein_prefix_choices = [
+            '01',
+            '02',
+            '03',
+            '04',
+            '05',
+            '06',
+            '10',
+            '11',
+            '12',
+            '13',
+            '14',
+            '15',
+            '16',
+            '20',
+            '21',
+            '22',
+            '23',
+            '24',
+            '25',
+            '26',
+            '27',
+            '30',
+            '31',
+            '32',
+            '33',
+            '34',
+            '35',
+            '36',
+            '37',
+            '38',
+            '39',
+            '40',
+            '41',
+            '42',
+            '43',
+            '44',
+            '45',
+            '46',
+            '47',
+            '48',
+            '50',
+            '51',
+            '52',
+            '53',
+            '54',
+            '55',
+            '56',
+            '57',
+            '58',
+            '59',
+            '60',
+            '61',
+            '62',
+            '63',
+            '64',
+            '65',
+            '66',
+            '67',
+            '68',
+            '71',
+            '72',
+            '73',
+            '74',
+            '75',
+            '76',
+            '77',
+            '80',
+            '81',
+            '82',
+            '83',
+            '84',
+            '85',
+            '86',
+            '87',
+            '88',
+            '90',
+            '91',
+            '92',
+            '93',
+            '94',
+            '95',
+            '98',
+            '99']
+
+        ein_prefix = random.choice(ein_prefix_choices)
+        sequence = self.random_int(min=0, max=9999999)
+
+        ein = "{0:s}-{1:07d}".format(ein_prefix, sequence)
+        return ein
+
+    def ssn(self, taxpayer_identification_number_type=SSN_TYPE):
+        """ Generate a random United States Taxpayer Identification Number of the specified type.
+
+        If no type is specified, a US SSN is returned.
+        """
+
+        if taxpayer_identification_number_type.lower() == self.ITIN_TYPE:
```

Comment:
let's skip the `lower()`s and make the argument case-sensitive, so that the exception message is accurate.

Reply:
I'm not convinced. Since SSN, ITIN, and EIN are initialisms/acronyms, it would be more correct to require these to be capitalized and not all lowercase (`taxpayer_identification_number_type in ['SSN', 'EIN', 'ITIN']`) but it seems unnecessarily strict to make the arguments case sensitive if there's no ambiguity in what the user meant.

Are you flexible on this one?

Reply:
we can make them uppercase, but I still want them to be case-sensitive, so the error message is 100% accurate. If it says "taxpayer_identification_number_type must be one of 'SSN', 'EIN', or 'ITIN'", then we must only accept those exact values. I'm ok with being overly strict if it means removing ambiguity.